attribute-service-client/src/main/java/org/hypertrace/core/attribute/service/client/AttributeServiceCachedClient.java

```diff
@@ -30,102 +31,91 @@
+            .maximumSize(clientConfig.getMaxSize())
+            .refreshAfterWrite(clientConfig.getRefreshAfterWrite())
+            .expireAfterWrite(clientConfig.getExpireAfterAccess())
             .build(
                 CacheLoader.asyncReloading(
                     CacheLoader.from(this::loadTable),
                     Executors.newFixedThreadPool(
-                        attributeServiceConfig.hasPath(CACHE_EXECUTOR_THREADS_CONFIG_KEY)
-                            ? attributeServiceConfig.getInt(CACHE_EXECUTOR_THREADS_CONFIG_KEY)
-                            : 4,
-                        this.buildThreadFactory())));
+                        clientConfig.getExecutorThreads(), this.buildThreadFactory())));
     PlatformMetricsRegistry.registerCache(
-        AttributeServiceCachedClient.class.getName(), cache, Collections.emptyMap());
+        clientConfig.getCacheMetricsName(), cache, Collections.emptyMap());
     scopeAndKeyLookup =
-        CacheBuilder.newBuilder().expireAfterWrite(expireAfterWriteDuration).build();
+        CacheBuilder.newBuilder().expireAfterWrite(clientConfig.getExpireAfterAccess()).build();
   }
 
   public Optional<AttributeMetadata> get(
-      String tenantId, String attributeScope, String attributeKey) {
+      @Nonnull RequestContext requestContext,
+      @Nonnull String attributeScope,
+      @Nonnull String attributeKey)
+      throws ExecutionException {
     try {
-      return Optional.ofNullable(cache.getUnchecked(tenantId))
+      return getTableForRequestContext(requestContext)
           .map(table -> table.get(attributeScope, attributeKey));
-    } catch (Exception e) {
-      if (LOG_LIMITER.tryAcquire()) {
-        log.error("No attribute available for scope {} and key {}", attributeScope, attributeKey);
-      }
+    } catch (NullPointerException e) {
+      log.debug("No attribute available for scope {} and key {}", attributeScope, attributeKey);
       return Optional.empty();
     }
   }
 
-  public Optional<AttributeMetadata> getById(String tenantId, String attributeId) {
+  public Optional<AttributeMetadata> getById(
+      @Nonnull RequestContext requestContext, @Nonnull String attributeId)
+      throws ExecutionException {
     try {
-      return Optional.ofNullable(cache.getUnchecked(tenantId))
-          .flatMap(
-              table ->
-                  Optional.ofNullable(scopeAndKeyLookup.getIfPresent(attributeId))
-                      .map(scopeAndKey -> table.get(scopeAndKey.getScope(), scopeAndKey.getKey())));
-    } catch (Exception e) {
-      if (LOG_LIMITER.tryAcquire()) {
-        log.error("No attribute available for id {}", attributeId);
-      }
+      return getTableForRequestContext(requestContext)
+          .map(
+              table -> {
+                AttributeScopeAndKey scopeAndKey = scopeAndKeyLookup.getIfPresent(attributeId);
+                return table.get(scopeAndKey.getScope(), scopeAndKey.getKey());
```

Comment:
Now here we could throw a NPE - I would suggest fixing that, should behave identically to the scope/key version